integrationTests/vm/wasm/wasmVM_test.go

```diff
@@ -0,0 +1,205 @@
+package arwen
+
+import (
+	"encoding/hex"
+	"fmt"
+	"io/ioutil"
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go/data/transaction"
+	"github.com/ElrondNetwork/elrond-go/integrationTests/vm"
+	"github.com/ElrondNetwork/elrond-go/process/factory"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestVmDeployWithTransferAndGasShouldDeploySCCode(t *testing.T) {
+	senderAddressBytes := []byte("12345678901234567890123456789012")
+	senderNonce := uint64(0)
+	senderBalance := big.NewInt(100000000)
+	round := uint64(444)
+	gasPrice := uint64(1)
+	gasLimit := uint64(100000)
+	transferOnCalls := big.NewInt(50)
+
+	scCode, err := ioutil.ReadFile("./store.wasm")
+	assert.Nil(t, err)
+
+	scCodeString := hex.EncodeToString(scCode)
+
+	tx := vm.CreateTx(
+		t,
+		senderAddressBytes,
+		vm.CreateEmptyAddress().Bytes(),
+		senderNonce,
+		transferOnCalls,
+		gasPrice,
+		gasLimit,
+		scCodeString+"@"+hex.EncodeToString(factory.HeraWABTVirtualMachine),
+	)
+
+	txProc, accnts, _ := vm.CreatePreparedTxProcessorAndAccountsWithVMs(t, senderNonce, senderAddressBytes, senderBalance)
+
+	err = txProc.ProcessTransaction(tx, round)
+	assert.Nil(t, err)
+
+	_, err = accnts.Commit()
+	assert.Nil(t, err)
+
+	expectedBalance := big.NewInt(99999805)
+	fmt.Printf("%s \n", hex.EncodeToString(expectedBalance.Bytes()))
+
+	vm.TestAccount(
+		t,
+		accnts,
+		senderAddressBytes,
+		senderNonce+1,
+		expectedBalance)
+}
+
+func Benchmark_VmDeployWithFibbonacciAndExecute(b *testing.B) {
+	runWASMVMBenchmark(b, "./fibonacci_ewasmified.wasm", b.N, 32)
+}
+
+func Benchmark_VmDeployWithCPUCalculateAndExecute(b *testing.B) {
+	runWASMVMBenchmark(b, "./cpucalculate_ewasmified.wasm", b.N, 8000)
+}
+
+func Benchmark_VmDeployWithStringConcatAndExecute(b *testing.B) {
+	runWASMVMBenchmark(b, "./stringconcat_ewasmified.wasm", b.N, 10000)
+}
+
+func runWASMVMBenchmark(tb testing.TB, fileSC string, numRun int, testingValue uint64) {
```

Comment:
This function runs a benchmark on WAVM specifically. Its name should rather be `runWAVMBenchmark()` to reflect this.

Reply:
done